backend/src/main/java/pl/edu/agh/awiteks_backend/services/PlantService.java

```diff
@@ -116,19 +128,39 @@ public PlantsStats getPlantsStats(int userId) {
     }
 
     public Plant updatePlant(AddPlantRequestBody addPlantRequestBody, int plantId, int userId) {
-        var plant = makePlantFromRequestBody(addPlantRequestBody, userId);
-        plant.setId(plantId);
-
-        plantRepository.save(plant);
-
-        return plant;
+        var plant = plantRepository
+                .findByIdAndUserId(plantId, userId)
+                .orElseThrow();
+
+            plant.setName(addPlantRequestBody.name());
+            plant.setActualInsolation(addPlantRequestBody.insolation());
+            plant.setNote(addPlantRequestBody.note());
+            // TODO handle all fields, remove activity if it was later than this or smth
+
+            if (!plant.getLastFertilizationDate().equals(addPlantRequestBody.lastFertilizationDate())) {
```

Comment:
Ten warunek można można wrzucić do osobnej funkcji i wywołać w `if`'ie

Reply:
Nie wiem czy o to chodziło ale rzeczywiście nie wyglądało to dobrze i poleciał refactor